subsys/bluetooth/host/crypto.c

```diff
@@ -114,6 +115,44 @@ int bt_rand(void *buf, size_t len)
 
 	return -EIO;
 }
+#else /* !CONFIG_BT_HOST_CRYPTO_PRNG */
+int bt_rand(void *buf, size_t len)
+{
+	int ret, size;
+	size_t i = 0;
+
+	/* Check first that HCI_LE_Rand is supported */
+	if (!BT_CMD_TEST(bt_dev.supported_commands, 27, 7)) {
+		return -ENOTSUP;
+	}
+
+	while (len) {
```

Comment:
This code duplicates the implementation of the entropy over HCI source:
https://github.com/zephyrproject-rtos/zephyr/pull/41424/files#diff-cd0469090601a75fd60f22e823813f55b66de8d019fd2203a33cb10b5bcee433R31

Wouldn't it make sense to combine bt_rand and the entropy over HCI into a single entity?

In a combined build using the entropy source for HCI is now less efficient than calling bt_rand, because bt_rand has the controller/crypto.c implementation.

To me it seems that bt_rand is still the best option for the application to use, but entropy source is the most portable option.
In my opinion it would make more sense to either:
- Entropy source over bt_rand instead of HCI.
- Deprecate bt_rand and use the entropy source in the host instead.

Reply:
cc @JordanYates

Reply:
Please replace this with the use of the `bt-hci` entropy driver:

```
entropy_driver = DEVICE_DT_GET(DT_CHOSEN(zephyr_entropy));
rc = entropy_get_entropy(entropy_driver, (uint8_t *)&state, sizeof(state));
```